pyo3-macros-backend/src/method.rs

```diff
@@ -47,26 +135,30 @@ impl<'a> FnArg<'a> {
                     other => return Err(handle_argument_error(other)),
                 };
 
-                let is_cancel_handle = arg_attrs.cancel_handle.is_some();
+                if utils::is_python(&cap.ty) {
+                    return Ok(Self::Py(PyArg {
+                        name: ident,
+                        ty: &cap.ty,
+                    }));
+                }
 
-                Ok(FnArg {
+                if arg_attrs.cancel_handle.is_some() {
+                    return Ok(Self::CancelHandle(CancelHandleArg {
```

Comment:
At this point I guess we should check that no other `arg_attrs` are set (else we'd silently ignore them)?

Reply:
True, looking at this again, there is also no reason to pass the whole `PyFunctionArgPyO3Attributes` into the `RegularArg` since only `from_py_with` is applicable there. I'll change that as well.

Reply:
It looks like `PyFunctionArgPyO3Attributes::from_attrs``already checks for incompatible attribute combinations, so I guess I'll just add a comment here?

Reply:
Yep that seems reasonable to me 👍